src/pages/index.js

```diff
@@ -19,64 +21,192 @@ import {
     },
   },
   ".cards__list"
 );
 
+const confirmDeleteModal = new PopupWithConfirm(
+  "#delete-modal",
+  handleDeleteClick
+);
+confirmDeleteModal.setEventListeners();
+
+const editAvatarModal = new PopupWithForm("#edit-avatar-modal", (input) => {
+  editAvatarModal.setLoading("Saving...");
+  handleAvatarEditSubmit(input);
+});
+editAvatarModal.setEventListeners();
+
+const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
+  const cardData = {
+    name: data.title,
+    link: data.url,
+  };
+
+  addCardPopup.setLoading("Saving...");
+
+  api
+    .addCards(cardData)
+    .then((res) => {
+      const card = createCard(res);
+      cardList.addItem(card);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+      addCardPopup.setLoading("Save");
+    })
+    .catch((err) => {
+      console.error(`Error, could not add card: ${err}`);
+    })
+    .finally(() => {
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.setLoading("Save");
+    });
+});
+
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
-const editFormValidator = new FormValidator(settings, profileEditForm);
-const addFormValidator = new FormValidator(settings, addCardFormElement);
+function handleLikeClick(data) {
+  if (!data.isLiked) {
+    api
+      .likeACard(data._id, { method: "PUT" })
+      .then((res) => {
+        console.log(res);
+        data.isLiked = true;
+        data.toggleLike();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .dislikeACard(data._id, { method: "DELETE" })
+      .then((res) => {
+        console.log(res);
+        data.isLiked = false;
+        data.toggleLike();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleDeleteClick(card) {
+  const cardId = card._id;
+  if (!cardId) {
+    console.error("card._id is undefined");
+    return;
+  }
+  confirmDeleteModal.setSubmitFunction(() => {
+    api
+      .handleDeleteCard(cardId)
+      .then((res) => {
+        console.log(res);
+        card.handleDeleteCard();
+        confirmDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  confirmDeleteModal.open();
+}
 
 function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleCardPreview);
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handleCardPreview,
+    handleLikeClick,
+    handleDeleteClick
+  );
   return cardElement.getView();
 }
 
 previewImagePopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#edit-modal",
-  handleProfileEditSubmit
-);
+const editProfilePopup = new PopupWithForm("#edit-modal", (userData) => {
+  editProfilePopup.setLoading("Saving...");
```

Comment:
`editProfilePopup.setLoading("Saving...")` should be inside `handleProfileEditSubmit` to make the code consistent


So, here will be only this:


```js

const editProfilePopup = new PopupWithForm("#edit-modal", handleProfileEditSubmit);

```